tests/test_products.py

```diff
@@ -534,3 +484,77 @@ def test_brew_tag_matching(variant_name, cpe, brew_tag, stream_name, requests_mo
     stream = ProductStream.objects.get(name=stream_name)
     assert stream.productvariants.get_queryset().count() == 0
     assert stream.cpes_from_brew_tags == [cpe]
+
+
+@pytest.mark.django_db
+@patch("corgi.tasks.prod_defs.slow_remove_product_from_build.delay")
+def test_multi_variant_streams(mock_remove, requests_mock):
+    """Test that errata_info variants are only attached to an active stream where that stream
+    shares errata_info with an inactive stream"""
+    with open("tests/data/prod_defs/proddefs-update-multi-stream-variant.json") as prod_defs:
+        text = prod_defs.read()
+        requests_mock.get(f"{settings.PRODSEC_DASHBOARD_URL}/product-definitions", text=text)
+
+    update_products()
+    assert not mock_remove.called
+
+    ga_stream = ProductStream.objects.get(name="ga_stream")
+    assert ga_stream.productvariants.get_queryset().count() == 1
+    z_stream = ProductStream.objects.get(name="z_stream")
+    assert z_stream.productvariants.get_queryset().count() == 0
+    assert z_stream.yum_repositories
+
+    sb = SoftwareBuildFactory()
+
+    ProductComponentRelation.objects.create(
+        external_system_id="https://example.com/rhel-8/compose/yum_stream/",
+        product_ref="z_stream",
+        software_build=sb,
+        type=ProductComponentRelation.Type.YUM_REPO,
+    )
+
+    with open("tests/data/prod_defs/proddefs-update-multi-stream-variant-update.json") as prod_defs:
+        text = prod_defs.read()
+        requests_mock.get(f"{settings.PRODSEC_DASHBOARD_URL}/product-definitions", text=text)
+
+    update_products()
+
+    ga_stream.refresh_from_db()
+    assert ga_stream.productvariants.get_queryset().count() == 0
+    z_stream.refresh_from_db()
+    assert z_stream.productvariants.get_queryset().count() == 1
+    assert not ProductComponentRelation.objects.filter(software_build=sb).exists()
+    assert mock_remove.called_with(
```

Comment:
Maybe below is better, or maybe it's expected that the mock is called multiple times?
```suggestion
    assert mock_remove.called_once_with(
```

Reply:
fixed.